<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>482</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Select Algorithm:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="algorithm">
              <item>
               <property name="text">
                <string>Genetic Algorithm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Hybrid Genetic Algorithm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Greedy Algorithm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Simulated Annealing</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Number of executions:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="execNum">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Check Time:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="checktime_yes">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Yes</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="checktime_no">
              <property name="text">
               <string>No</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>File with algorithm settings:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLineEdit" name="algconfname"/>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton">
                <property name="text">
                 <string>...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>System config file:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLineEdit" name="sysconfname"/>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton_2">
                <property name="text">
                 <string>...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Result filename:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="result_filename"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>System Configuration:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Min Cost:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="mincost">
                <property name="text">
                 <string>--</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Max Cost:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="maxcost">
                <property name="text">
                 <string>--</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Cost Limit:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="limitcost"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Min Times:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="mintime">
                <property name="text">
                 <string>--</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Max Times:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="maxtime">
                <property name="text">
                 <string>--</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Times Limits:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="limittimes"/>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <spacer name="horizontalSpacer_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_2">
              <property name="text">
               <string>Random</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_3">
              <property name="text">
               <string>Save as ...</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_8">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="use_metamodels">
            <property name="text">
             <string>Use metamodels</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Metamodel:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="metamodel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <item>
             <property name="text">
              <string>Averaging</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>KNearestNeighbours</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Neural Network</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Support Vector Regression</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Linear Regression</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Random</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Population control percent:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="popControl">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Run</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="metamodels_button">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/pics/pics/chart_line.png</normaloff>:/pics/pics/chart_line.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>482</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as ...</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionEdit">
   <property name="text">
    <string>Edit</string>
   </property>
  </action>
  <action name="actionSave_2">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_as_2">
   <property name="text">
    <string>Save as ...</string>
   </property>
  </action>
  <action name="actionOpen_2">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionEdit_2">
   <property name="text">
    <string>Edit</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>264</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OpenAlgConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>192</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OpenSysConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>192</x>
     <y>311</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>algconfname</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>LoadAlgConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sysconfname</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>LoadSysConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Random()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>481</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>SaveSysConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>369</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>limitcost</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>InputCostLimit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>501</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>limittimes</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>InputTimeLimits()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checktime_no</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>no_checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>-17</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checktime_yes</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>yes_checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>-21</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>use_metamodels</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>LoadSysConf()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>265</y>
    </hint>
    <hint type="destinationlabel">
     <x>-42</x>
     <y>127</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>use_metamodels</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>use_metamodels_checked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>metamodels_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowMetamodelRes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>271</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>432</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Run()</slot>
  <slot>OpenAlgConf()</slot>
  <slot>OpenSysConf()</slot>
  <slot>LoadSysConf()</slot>
  <slot>LoadAlgConf()</slot>
  <slot>Random()</slot>
  <slot>InputTimeLimits()</slot>
  <slot>InputCostLimit()</slot>
  <slot>SaveSysConf()</slot>
  <slot>no_checked()</slot>
  <slot>yes_checked()</slot>
  <slot>use_metamodels_checked()</slot>
  <slot>ShowMetamodelRes()</slot>
 </slots>
</ui>
